verifier/src/report_body.rs

```diff
@@ -59,167 +59,129 @@ report_body_field_accessor! {
-            VerificationError::ConfigIdMismatch { expected, actual },
-            is_some.into(),
-        )
+        CtOption::new(T::into_verification_error(expected, actual), is_some.into())
     }
 }
 
-/// Verify the [`ConfigSvn`] is as expected.
+/// Common implementation for [`Verifier`]s that test for an actual value being
+/// greater than or equal to an expected value
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
-pub struct ConfigSvnVerifier {
-    expected_svn: ConfigSvn,
+pub struct GreaterThanEqualVerifier<T> {
+    expected: T,
 }
 
-impl ConfigSvnVerifier {
-    /// Create a new [`ConfigSvnVerifier`] instance.
-    ///
-    /// Verifies that `expected_svn` is less than the svn found in evidence
-    /// during `verify()`.
-    pub fn new(expected_svn: ConfigSvn) -> Self {
-        Self { expected_svn }
+impl<T> GreaterThanEqualVerifier<T> {
+    pub fn new(expected: T) -> Self {
+        Self { expected }
     }
 }
 
-impl<T: Accessor<ConfigSvn>> Verifier<T> for ConfigSvnVerifier {
-    type Error = VerificationError;
-
-    fn verify(&self, evidence: &T) -> CtOption<Self::Error> {
-        let expected = self.expected_svn.clone();
-        let actual = evidence.get();
-
-        let is_some = actual.as_ref().ct_lt(expected.as_ref());
-        CtOption::new(
-            VerificationError::ConfigSvnTooSmall { expected, actual },
-            is_some,
-        )
+/// Verifier for ensuring [`Attributes`] values are equivalent.
+pub type AttributesVerifier = EqualityVerifier<Attributes>;
+impl IntoVerificationError for Attributes {
+    fn into_verification_error(expected: Self, actual: Self) -> VerificationError {
+        VerificationError::AttributeMismatch { expected, actual }
     }
 }
 
-/// Verify the [`IsvSvn`] is as expected.
-#[derive(Clone, Debug, Eq, Hash, PartialEq)]
-pub struct IsvSvnVerifier {
-    expected_svn: IsvSvn,
+/// Verifier for ensuring [`ConfigId`] values are equivalent.
+pub type ConfigIdVerifier = EqualityVerifier<ConfigId>;
+impl IntoVerificationError for ConfigId {
+    fn into_verification_error(expected: Self, actual: Self) -> VerificationError {
+        VerificationError::ConfigIdMismatch { expected, actual }
+    }
 }
 
-impl IsvSvnVerifier {
-    /// Create a new [`IsvSvnVerifier`] instance.
-    ///
-    /// Verifies that `expected_svn` is less than the svn found in evidence
-    /// during `verify()`.
-    pub fn new(expected_svn: IsvSvn) -> Self {
-        Self { expected_svn }
+/// Verifier for ensuring [`ConfigSvn`] is greater than or equal to an
+/// expected [`ConfigSvn`]
+pub type ConfigSvnVerifier = GreaterThanEqualVerifier<ConfigSvn>;
+impl IntoVerificationError for ConfigSvn {
+    fn into_verification_error(expected: Self, actual: Self) -> VerificationError {
+        VerificationError::ConfigSvnTooSmall { expected, actual }
     }
 }
 
-impl<T: Accessor<IsvSvn>> Verifier<T> for IsvSvnVerifier {
+impl<E: Accessor<ConfigSvn>> Verifier<E> for GreaterThanEqualVerifier<ConfigSvn> {
```

Comment:
manually implementing for each version, until #280 comes in, added a task to #27 to ensure it doesn't get lost.